### PR TITLE
Connect: Make sure tsh auto-updates are turned off

### PR DESCRIPTION
**Pull request by ravicious**
This PR makes it so that tsh never attempts to auto-update itself when it's executed by Connect, [as described in RFD 144](https://github.com/gravitational/teleport/blob/c906170e1dc2c7f32a4902c4ba81db2e29efb654/rfd/0144-client-tools-updates.md#out-of-scope).

I think I haven't missed any place. I verified that this works by starting Connect, starting an SSH session and a local terminal and then running `ps eww <pid> | rg TELEPORT` on the tsh daemon process and the `tsh ssh` process. I also verified that the env var is present in the local terminal session.

I also verified that the env var is set correctly on Windows in cmd.exe, pwsh.exe and wsl.exe and that the launch daemon receives the env var.

### CODE REVIEW

File: web/packages/teleterm/src/node/README.md
Comment (ravicious) [inline web/packages/teleterm/src/node/README.md:1]: @gzdunek I added this folder so we can start sorting shared files. I imagine we'd also want to create a folder called "universal" for files that are used both in the browser and in Node.js.

File: build.assets/macos/tsh/tsh.app/Contents/Library/LaunchDaemons/com.gravitational.teleport.tsh.vnetd.plist [hunk 1]
```diff
@@ -22,5 +22,12 @@
 	<string>/var/log/vnet.log</string>
 	<key>ThrottleInterval</key>
 	<integer>5</integer>
+	<key>EnvironmentVariables</key>
+	<dict>
+		<!-- Auto-updates are disabled because $HOME isn't available,
+		but let's also disable them explicitly anyway. -->
+		<key>TELEPORT_TOOLS_VERSION</key>
+		<string>off</string>
```
Comment (ravicious) [inline build.assets/macos/tsh/tsh.app/Contents/Library/LaunchDaemons/com.gravitational.teleport.tsh.vnetd.plist:30]: The env vars available in the launch daemon process are just these:

```
TELEPORT_TOOLS_VERSION=off XPC_SERVICE_NAME=com.goteleport.tshdev.vnetd PATH=/usr/bin:/bin:/usr/sbin:/sbin XPC_FLAGS=1
```

**Review by avatus — APPROVED** [review]

File: web/packages/teleterm/src/node/README.md
Comment (gzdunek) [reply web/packages/teleterm/src/node/README.md:1]: Good idea.

**Review by gzdunek — APPROVED** [review]

**Comment by ravicious** [comment]
Backported to v16 and v15 here:

* https://github.com/gravitational/teleport/pull/48645
* https://github.com/gravitational/teleport/pull/48648

**Pull request merged**